components/VM_Arm/configurations/vm.h

```diff
@@ -68,22 +68,28 @@
     attribute int num_extra_frame_caps; \
     attribute int extra_frame_map_address; \
     attribute { \
-        string linux_ram_base; \
-        string linux_ram_paddr_base; \
-        string linux_ram_size; \
-        string linux_ram_offset; \
+        string ram_base; \
+        string ram_paddr_base; \
+        string ram_size; \
+        string ram_offset; \
         string dtb_addr; \
         string initrd_max_size; \
         string initrd_addr; \
```

Comment:
It seems like there's an easy way to implement these configuration improvements without breaking any existing project CAmkES configurations. (There's lots of projects, examples, tutorials, and old emails sitting around that use these configs). If we defined vm_address_config and vm_image_config as new attributes while leaving the existing attribute maps around then new projects could choose to use the new config types and old projects and docs will continue to work.

The VM component can then require that if any of the new settings are set, the old settings must be unset or something.

Reply:
Good idea. I'll add that to my "dereference linux" commit. I'll put some warning about using a deprecated option.

Reply:
If we keep compatibility, then the new values should really move from `string` to `uint64_t` for all new address and offset attributes. There seem no reason to keep them as strings.

Reply:
I'm using templates to create a file containing the necessary values. It can pull from `vm_image_config` or `linux_image_config`. So its fine to leave them as strings, because the end result is this:

```
#warning You are using the deprecated linux_address_config structure. Please use the vm_address_config structure instead
#warning You are using the deprecated linux_image_config structure. Please use the vm_image_config structure instead

unsigned long ram_base = 0x40000000;
unsigned long ram_paddr_base = 0x40000000;
unsigned long ram_size = 0x20000000;
unsigned long ram_offset = 0x00000000;
unsigned long dtb_addr = 0x4F000000;
unsigned long initrd_max_size = 0x1900000;
unsigned long initrd_addr = 0x4D700000;
```

Reply:
Thanks for addressing this request so late in the review cycle @chrisguikema !

Reply:
No worries, Kent. I think this feature would help out a lot of people, and it gets DornerWorks one step closer to having all our stuff open-sourced, so I'm happy to make the changes necessary to get this merged.